mlrun/datastore/sources.py

```diff
@@ -412,6 +413,51 @@ def to_dataframe(
             **reader_args,
         )
 
+    def _get_spark_additional_filters(self):
+        if not self.additional_filters:
+            return None
+        none_values = [None, np.nan, np.NaN]
+        additional_filters = deepcopy(self.additional_filters)
+        is_first = True
+        spark_filter = ""
+        for filter_tuple in additional_filters:
+            if not filter_tuple:
+                continue
+            if not is_first:
+                spark_filter += " AND "
+            else:
+                is_first = False
+            col_name, op, value = filter_tuple
+            if isinstance(value, (str, datetime)):
+                value = rf"'{value}'"
+            if isinstance(value, (list, tuple, set, np.ndarray)) and (
+                "in" in op.lower()
+            ):
+                none_exists = False
+                value = list(value)
+                for none_value in none_values:
+                    if none_value in value:
+                        value.remove(none_value)
+                        none_exists = True
+                value = tuple(value)
+                if len(value) == 1:
+                    value = f"({value[0]})"
+                if none_exists:
+                    if value:
+                        spark_filter += f"({str(col_name)} {str(op)} {str(value)} OR {str(col_name)} IS NULL)"
```

Comment:
This is actually not really safe... It opens us up to (probably accidental rather than malicious) SQL injection. It would be better to use the structured API – https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.filter.html (see "Filter by Column instances").

Reply:
`a`